### PR TITLE
Pass type through FlattenedSchemaRequest to correctly handle nullability

### PR DESCRIPTION
**Pull request by chamons**
- Fixes https://github.com/azure/autorest.csharp/issues/1054 (At least
  the part that is invalid code and not a feature)

# Checklist

To ensure a quick review and merge, please ensure:
- [x] The PR has a understandable title and description explaining the _why_ and _what_.
- [x] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [x] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/AutoRest.CSharp/Properties/launchSettings.json [hunk 1]
```diff
@@ -172,6 +172,10 @@
       "commandName": "Project",
       "commandLineArgs": "--standalone $(SolutionDir)\\test\\TestProjects\\ExtensionClientName\\Generated"
     },
+    "FlatArray": {
```
Comment (pakrym) [inline src/AutoRest.CSharp/Properties/launchSettings.json:175]: Let's make this project name less specific. FlattenedParameters?

File: test/AutoRest.TestServer.Tests/flat-array.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            using var testServer = new InProcTestServer(async content =>
+            {
+                await content.Request.Body.CopyToAsync(requestMemoryStream);
+            });
+
+            var response = await new FlatArrayClient(ClientDiagnostics, HttpPipeline, testServer.Address).OperationAsync();
+
+            var doc = JsonDocument.Parse(Encoding.UTF8.GetString(requestMemoryStream.ToArray()));
+            JsonElement items = doc.RootElement.GetProperty("items");
+            Assert.NotNull (items);
```
Comment (pakrym) [inline test/AutoRest.TestServer.Tests/flat-array.cs:30]: nit spaces.

**Comment by pakrym** [comment]
>Fixes #1054

We still a null check for non-nullable collections.

**Review by pakrym — APPROVED** [review]

File: src/AutoRest.CSharp/Properties/launchSettings.json (hunk 1 above)
Comment (chamons) [reply src/AutoRest.CSharp/Properties/launchSettings.json]: Done

**Review by pakrym — APPROVED** [review]

**Pull request merged**